packages/in_app_purchase/example/ios/in_app_purchase_pluginTests/ProductRequestHandlerTest.m

```diff
@@ -0,0 +1,180 @@
+//
+//  ProductRequestHandlerTest.m
+//  in_app_purchase_pluginTests
+//
+//  Created by Chris Yang on 1/11/19.
+//  Copyright © 2019 The Chromium Authors. All rights reserved.
+//
+
+#import <XCTest/XCTest.h>
+#import "FLTSKProductRequestHandler.h"
+
+#pragma stubs
+
+@interface SKProductSubscriptionPeriodStub : SKProductSubscriptionPeriod
+@end
+
+@implementation SKProductSubscriptionPeriodStub
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self setValue:@(0) forKey:@"numberOfUnits"];
+    [self setValue:@(0) forKey:@"unit"];
+  }
+  return self;
+}
+
+@end
+
+@interface SKProductDiscountStub : SKProductDiscount
+@end
+
+@implementation SKProductDiscountStub
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self setValue:@(1.0) forKey:@"price"];
+    NSLocale *locale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US"];
+    [self setValue:locale forKey:@"priceLocale"];
+    [self setValue:@(1) forKey:@"numberOfPeriods"];
+    SKProductSubscriptionPeriodStub *subscriptionPeriodSub =
+        [[SKProductSubscriptionPeriodStub alloc] init];
+    [self setValue:subscriptionPeriodSub forKey:@"subscriptionPeriod"];
+    [self setValue:@(1) forKey:@"paymentMode"];
+  }
+  return self;
+}
+
+@end
+
+@interface SKProductStub : SKProduct
+@end
+
+@implementation SKProductStub
+
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    [self setValue:@"consumable" forKey:@"productIdentifier"];
+    [self setValue:@"description" forKey:@"localizedDescription"];
+    [self setValue:@"title" forKey:@"localizedTitle"];
+    [self setValue:@YES forKey:@"downloadable"];
+    [self setValue:@(1.0) forKey:@"price"];
+    NSLocale *locale = [[NSLocale alloc] initWithLocaleIdentifier:@"en_US"];
+    [self setValue:locale forKey:@"priceLocale"];
+    [self setValue:@[ @1, @2 ] forKey:@"downloadContentLengths"];
+    SKProductSubscriptionPeriodStub *period = [[SKProductSubscriptionPeriodStub alloc] init];
+    [self setValue:period forKey:@"subscriptionPeriod"];
+    SKProductDiscountStub *discount = [[SKProductDiscountStub alloc] init];
+    [self setValue:discount forKey:@"introductoryPrice"];
+    [self setValue:@"com.group" forKey:@"subscriptionGroupIdentifier"];
+  }
+  return self;
+}
+
+@end
+
+#pragma tests start here
+
+@interface ProductRequestHandlerTest : XCTestCase
+
+@end
+
+@implementation ProductRequestHandlerTest
+
+- (void)testSKProductSubscriptionPeriodStubToMap {
+  SKProductSubscriptionPeriodStub *period = [[SKProductSubscriptionPeriodStub alloc] init];
+  NSDictionary *map = [period toMap];
+  NSDictionary *match = @{@"numberOfUnits" : @(0), @"unit" : @(0)};
```

Comment:
nit: is It possible to clearly create the `match` and `notMatch` maps throughout here using the values from the stub? It would make the tests less brittle because they'd be actually using the stub values for the assertions instead of having them hardcoded.

Something like:

```
*match = @{@"numberOfUnits" : [period numberOfUnits], @"unit" : [period unit]};
*notMatch = @{@"numberOfUnits" : [period numberOfUnits] + 1, @"unit" : @(1)};
```